### PR TITLE
telemetry,session: add telemetry for the temporary table feature

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Add telemetry for the temporary table.

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Check all the tables meta info to see is there any temporary table exists.

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Review by wjhuang2016 — COMMENTED** [review]
If we want to know if a user uses the temporary table, we can use a variable to record it. When creating a temporary, mark it as true. For this PR's approach, the temporary table may be dropped. when scanning the infoschema, we can't find it. Besides, if a cluster reports `true`, and next time reports `false`, it's strange.

File: session/session.go [hunk 1]
```diff
@@ -2877,6 +2877,21 @@ func (s *session) GetTxnWriteThroughputSLI() *sli.TxnWriteThroughputSLI {
 	return &s.txn.writeSLI
 }
 
+var _ telemetry.TemporaryTableFeatureChecker = &session{}
+
+// TemporaryTableExists is used by the telemetry package to avoid circle dependency.
+func (s *session) TemporaryTableExists() bool {
+	is := domain.GetDomain(s).InfoSchema()
```
Comment (xhebox) [inline session/session.go:2884]: I would recommend `s.GetInfoSchema()` if your intention also includes history/snapshot/transaction cases. It is introduced by me to get the correct infoschema by the context.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Comment by tiancaiamao** [comment]
> the temporary table may be dropped. when scanning the infoschema, we can't find it. 
> Besides, if a cluster reports true, and next time reports false, it's strange.

I think that's OK.
If a temporary table is created, and then dropped, we view the case as **not used**.
The telemetry is not about "we provide a feature or not", it is about whether the user really **using this feature**.

> If we want to know if a user uses the temporary table, we can use a variable to record it. When creating a temporary, mark it as true. 

A variable can not survive the server restart scenario.
If you really want to know whether a user **has ever created** a temporary table, the variable has to be persistent to some non-volatile storage.

File: session/session.go (hunk 1 above)
Comment (tiancaiamao) [reply session/session.go]: Here it **should be** the latest schema, rather than a snapshot.

File: session/session.go [hunk 2]
```diff
@@ -2877,6 +2877,21 @@ func (s *session) GetTxnWriteThroughputSLI() *sli.TxnWriteThroughputSLI {
 	return &s.txn.writeSLI
 }
 
+var _ telemetry.TemporaryTableFeatureChecker = &session{}
+
+// TemporaryTableExists is used by the telemetry package to avoid circle dependency.
+func (s *session) TemporaryTableExists() bool {
```
Comment (djshow832) [inline session/session.go:2883]: How often is it called? It seems time-consuming if it's called frequently.

File: session/session.go [hunk 3]
```diff
@@ -2877,6 +2877,21 @@ func (s *session) GetTxnWriteThroughputSLI() *sli.TxnWriteThroughputSLI {
...
+	is := s.GetInfoSchema()
+	for _, dbInfo := range is.AllSchemas() {
+		for _, tbInfo := range is.SchemaTables(dbInfo.Name) {
```
Comment (djshow832) [inline session/session.go:2886]: Can you filter system databases by `IsMemOrSysDB`?

Comment (tiancaiamao) [reply session/session.go:2886]: Nope, we can create global temporary table under any DB like `mysql` ...

File: session/session.go (hunk 2 above)
Comment (tiancaiamao) [reply session/session.go:2883]: Every 6 hours...

Comment (tiancaiamao) [reply session/session.go:2883]: https://github.com/pingcap/tidb/blob/6e9d2750ae2042fb01acabb44de627ba92d7a1f4/domain/domain.go#L1054
and https://github.com/pingcap/tidb/blob/6e9d2750ae2042fb01acabb44de627ba92d7a1f4/telemetry/telemetry.go#L39

**Pull request merged**